### PR TITLE
fix(onyx-396): fixes search pagination issue when previous search results appear

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Fix**

This PR solves [ONYX-396]

### Description

Steps to reproduce:

- Type "kaws" in search input, press enter
- Type "warhol" in search input, press enter
- Under artworks grid, click on the "next" link

Outcome before this fix:

- Kaws's artworks are displayed in the grid

Expected outcome:

- Warhol's artworks should be displayed

[ONYX-396]: https://artsyproduct.atlassian.net/browse/ONYX-396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by nickskalkin** [comment]
props to @damassi for the implementation suggestion 🙏

**Review by dblandin — APPROVED** [review]
Awesome 💯

**Comment by MounirDhahri** [comment]
Thanks for the fix

File: src/Apps/Search/Routes/SearchResultsArtworks.tsx [hunk 1]
```diff
@@ -19,11 +20,20 @@ interface SearchResultsRouteProps {
...
   const { sidebar } = viewer
 
+  useEffect(() => {
+    // refresh artwork filter on query change
+    setSearchFilterKey(match.location.query.term)
+  }, [match.location.query.term])
+
   return (
     <ArtworkFilter
+      key={searchFitlerKey}
```
Comment (anandaroop) [inline src/Apps/Search/Routes/SearchResultsArtworks.tsx:36]: **question:** It would not have occurred to me to use a `key` outside of a list context; would be helpful to understand the rationale for this. I take it that this forces the component to essentially be remounted when the url (and therefore the key) changes, but why is this necessary, especially given the presence of L39 below? 🤔

**Review by anandaroop — COMMENTED** [review]
Thanks for fixing @nickskalkin  (and for the suggestion @damassi).

Just one q below, to better understand what's going on here…

File: src/Apps/Search/Routes/SearchResultsArtworks.tsx [hunk 2]
```diff
@@ -19,11 +20,20 @@ interface SearchResultsRouteProps {
 export const SearchResultsArtworksRoute: React.FC<SearchResultsRouteProps> = props => {
   const { match } = useRouter()
   const { userPreferences } = useSystemContext()
+  const [searchFitlerKey, setSearchFilterKey] = useState(
```
Comment (joeyAghion) [inline src/Apps/Search/Routes/SearchResultsArtworks.tsx:23]: Typo

File: src/Apps/Search/Routes/SearchResultsArtworks.tsx (hunk 1 above)
Comment (damassi) [reply src/Apps/Search/Routes/SearchResultsArtworks.tsx:36]: When you reset the key, it will also also refreshes all internal state. No new key, state internal to components via `useState` is preserved across renders. But as soon as a parent key changes, each of those calls get recomputed. Note that this can be expensive. 

There's the question of whether `match.location.subFooVariables` as returned from `useRouter` is internally triggering its own state change when those values update. I would expect this to be the case, as you mention @anandaroop! but maybe that doesn't.
